--- src/audacity.desktop.orig	2008-05-14 14:07:01.000000000 +0100
+++ src/audacity.desktop	2008-05-14 14:08:52.000000000 +0100
@@ -1,8 +1,6 @@
 [Desktop Entry]
-Encoding=UTF-8
-
-Name=Audacity Sound Editor
-Name[de]=Audacity Audio-Editor
+Name=Audacity
+Name[de]=Audacity
 Name[ru]=Audacity
 GenericName=Sound Editor
 GenericName[de]=Audio-Editor
@@ -11,10 +9,10 @@
 Comment[de]=Audio-Dateien aufnehmen und bearbeiten
 Comment[ru]=Запись и редактирование звуковых файлов
 
-Icon=/usr/share/audacity/audacity.xpm
+Icon=audacity
 
 Type=Application
-Categories=Application;AudioVideo;Audio;AudioVideoEditing;
+Categories=AudioVideo;Audio;AudioVideoEditing;
 
 Exec=audacity
 StartupNotify=false
