diff -Nur -x '*.orig' -x '*~' debian.git/src/FileNames.cpp debian.git.new/src/FileNames.cpp
--- debian.git/src/FileNames.cpp	2009-07-17 21:42:24.341690000 +0200
+++ debian.git.new/src/FileNames.cpp	2009-08-02 00:12:53.238525538 +0200
@@ -120,7 +120,7 @@
 {
    // Use the Audacity.exe directory as a base (on Windows, others elsewhere)
    wxString exeDir = wxStandardPaths::Get().GetDataDir();
-   return FileNames::MkDir( wxFileName( exeDir+wxT("/help/manual"), wxEmptyString ).GetFullPath() );
+   return wxFileName( exeDir+wxT("/help/manual"), wxEmptyString ).GetFullPath();
 }
 
 wxString FileNames::HtmlHelpIndexFile(bool quick)
