Description: Support FFmpeg >= 0.6
Bug: http://bugzilla.audacityteam.org/show_bug.cgi?id=176
Bug-Debian: http://bugs.debian.org/593162
Bug-Ubuntu: https://launchpad.net/bugs/602934
Author: Nicola Ferralis <feranick@hotmail.com>

diff -Nru audacity-src-1.3.12-beta.orig/src/FFmpeg.cpp audacity-src-1.3.12-beta/src/FFmpeg.cpp
--- audacity-src-1.3.12-beta.orig/src/FFmpeg.cpp	2010-03-30 04:28:54.000000000 -0700
+++ audacity-src-1.3.12-beta/src/FFmpeg.cpp	2010-09-03 16:05:44.505148215 -0700
@@ -356,7 +356,7 @@
          }
          // Otherwize, resort to extension matching if available
          else if (fmt1->extensions) {
-            if (FFmpegLibsInst->match_ext(filename, fmt1->extensions)) {
+            if (FFmpegLibsInst->av_match_ext(filename, fmt1->extensions)) {
                score = 50;
             }
          }
@@ -829,7 +829,7 @@
    INITDYN(avformat,av_open_input_file);
    INITDYN(avformat,av_open_input_stream);
    INITDYN(avformat,get_buffer);
-   INITDYN(avformat,match_ext);
+   INITDYN(avformat,av_match_ext);
 
 #if FFMPEG_STABLE
    INITDYN(avformat,av_init_packet);
diff -Nru audacity-src-1.3.12-beta.orig/src/FFmpeg.h audacity-src-1.3.12-beta/src/FFmpeg.h
--- audacity-src-1.3.12-beta.orig/src/FFmpeg.h	2010-03-30 04:28:54.000000000 -0700
+++ audacity-src-1.3.12-beta/src/FFmpeg.h	2010-09-03 16:06:32.413147835 -0700
@@ -224,7 +224,7 @@
    AVStream*         (*av_new_stream)                 (AVFormatContext *s, int id);
    AVFormatContext*  (*av_alloc_format_context)       (void);
    AVOutputFormat*   (*guess_format)                  (const char *short_name, const char *filename, const char *mime_type);
-   int               (*match_ext)                     (const char *filename, const char *extensions);
+   int               (*av_match_ext)                  (const char *filename, const char *extensions);
    int               (*av_write_trailer)              (AVFormatContext *s);
    int               (*av_interleaved_write_frame)    (AVFormatContext *s, AVPacket *pkt);
    int               (*av_write_frame)                (AVFormatContext *s, AVPacket *pkt);
