Description: Fix crash on WMA export.
Author: Leland Lucius <leland@audacityteam.org>
Bug: http://bugzilla.audacityteam.org/show_bug.cgi?id=274
Bug-Ubuntu: https://launchpad.net/bugs/695648

--- a/src/export/ExportFFmpeg.cpp
+++ b/src/export/ExportFFmpeg.cpp
@@ -591,13 +591,13 @@
       }
    }
 
+   // Write any file trailers.
+   FFmpegLibsInst->av_write_trailer(mEncFormatCtx);
+
    // Close the codecs.
    if (mEncAudioStream != NULL)
       FFmpegLibsInst->avcodec_close(mEncAudioStream->codec);
 
-   // Write any file trailers.
-   FFmpegLibsInst->av_write_trailer(mEncFormatCtx);
-
    for (i = 0; i < (int)mEncFormatCtx->nb_streams; i++)
    {
       FFmpegLibsInst->av_freep(&mEncFormatCtx->streams[i]->codec);
