Description: Fix an autoconf warning.
Author: Benjamin Drung <bdrung@debian.org>

--- a/configure.in
+++ b/configure.in
@@ -286,7 +286,7 @@
 dnl CXXFLAGS="$CXXFLAGS -I\$(top_srcdir)/lib-src/allegro"
 
 dnl Include FileDialog
-AC_CONFIG_SUBDIRS(["lib-src/FileDialog"])
+AC_CONFIG_SUBDIRS([lib-src/FileDialog])
 LOCAL_LIBS="$LOCAL_LIBS FileDialog.a"
 CXXFLAGS="$CXXFLAGS -I\$(top_srcdir)/lib-src/FileDialog"
 
