Description: respect locales/$LANG
Bug-Debian: http://bugs.debian.org/481424
Bug-Ubuntu: https://launchpad.net/bugs/292168
Forwarded: yes
Author: Benjamin Drung <bdrung@ubuntu.com>
diff --git a/src/AudacityApp.cpp b/src/AudacityApp.cpp
index 21be5ac..de36ecc 100644
--- a/src/AudacityApp.cpp
+++ b/src/AudacityApp.cpp
@@ -70,6 +70,7 @@ It handles initialization and termination by subclassing wxApp.
 #include "GStreamerLoader.h"
 #include "Internat.h"
 #include "LangChoice.h"
+#include "Languages.h"
 #include "Prefs.h"
 #include "Project.h"
 #include "Screenshot.h"
@@ -1024,9 +1025,8 @@ bool AudacityApp::OnInit()
 
    wxString lang = gPrefs->Read(wxT("/Locale/Language"), wxT(""));
 
-   // Pop up a dialog the first time the program is run
    if (lang == wxT(""))
-      lang = ChooseLanguage(NULL);
+      lang = GetSystemLanguageCode();
 
 #ifdef NOT_RQD
 //TIDY-ME: (CleanSpeech) Language prompt??
@@ -1038,7 +1038,6 @@ bool AudacityApp::OnInit()
 //lda   if (lang == "")
 //lda      lang = ChooseLanguage(NULL);
 #endif
-   gPrefs->Write(wxT("/Locale/Language"), lang);
 
    mLocale = NULL;
    InitLang( lang );
diff --git a/src/Languages.cpp b/src/Languages.cpp
index 061beb6..c625243 100644
--- a/src/Languages.cpp
+++ b/src/Languages.cpp
@@ -223,6 +223,10 @@ void GetLanguages(wxArrayString &langCodes, wxArrayString &langNames)
 
    tempNames.Sort();
 
+   // Add system language
+   langNames.Add(wxT("System"));
+   langCodes.Add(wxT(""));
+
    for(j=0; j<tempNames.GetCount(); j++) {
       langNames.Add(tempNames[j]);
       langCodes.Add(reverseHash[tempNames[j]]);
diff --git a/src/prefs/EffectsPrefs.cpp b/src/prefs/EffectsPrefs.cpp
index ce63d66..7728baf 100644
--- a/src/prefs/EffectsPrefs.cpp
+++ b/src/prefs/EffectsPrefs.cpp
@@ -126,6 +126,8 @@ bool EffectsPrefs::Apply()
 
    // If language has changed, we want to change it now, not on the next reboot.
    wxString lang = gPrefs->Read(wxT("/Locale/Language"), wxT(""));
+   if (lang == wxT(""))
+      lang = GetSystemLanguageCode();
    wxGetApp().InitLang(lang);
 
    return true;
diff --git a/src/prefs/GUIPrefs.cpp b/src/prefs/GUIPrefs.cpp
index 5971088..a6539c1 100644
--- a/src/prefs/GUIPrefs.cpp
+++ b/src/prefs/GUIPrefs.cpp
@@ -113,7 +113,7 @@ void GUIPrefs::PopulateOrExchange(ShuttleGui & S)
 
          S.TieChoice(_("&Language:"),
                      wxT("/Locale/Language"),
-                     wxT("en"),
+                     wxT(""),
                      mLangNames,
                      mLangCodes);
          S.SetSizeHints(mLangNames);
@@ -161,6 +161,8 @@ bool GUIPrefs::Apply()
 
    // If language has changed, we want to change it now, not on the next reboot.
    wxString lang = gPrefs->Read(wxT("/Locale/Language"), wxT(""));
+   if (lang == wxT(""))
+      lang = GetSystemLanguageCode();
    wxGetApp().InitLang(lang);
 
    return true;
