diff -Nur -x '*.orig' -x '*~' debian.git/lib-src/libnyquist/Makefile.in debian.git.new/lib-src/libnyquist/Makefile.in
--- debian.git/lib-src/libnyquist/Makefile.in	2009-06-14 15:40:18.000000000 +0200
+++ debian.git.new/lib-src/libnyquist/Makefile.in	2009-06-14 15:41:54.803169276 +0200
@@ -520,4 +520,5 @@
 	rm -f points.dat
 
 distclean: cleaner
+	-$(MAKE) -C misc distclean
 	rm -f Makefile
diff -Nur -x '*.orig' -x '*~' debian.git/lib-src/libnyquist/misc/Makefile.in debian.git.new/lib-src/libnyquist/misc/Makefile.in
--- debian.git/lib-src/libnyquist/misc/Makefile.in	2009-06-14 15:40:18.000000000 +0200
+++ debian.git.new/lib-src/libnyquist/misc/Makefile.in	2009-06-14 15:41:54.803169276 +0200
@@ -65,3 +65,5 @@
 	rm -f *.BAK
 	rm -f *~
 
+distclean: cleaner
+	rm -f Makefile
diff -Nur -x '*.orig' -x '*~' debian.git/lib-src/libvamp/Makefile.in debian.git.new/lib-src/libvamp/Makefile.in
--- debian.git/lib-src/libvamp/Makefile.in	2009-06-14 15:40:18.000000000 +0200
+++ debian.git.new/lib-src/libvamp/Makefile.in	2009-06-14 15:41:54.803169276 +0200
@@ -248,7 +248,7 @@
 		rm -f $(SDK_OBJECTS) $(HOSTSDK_OBJECTS) $(PLUGIN_OBJECTS) $(HOST_OBJECTS) $(RDFGEN_OBJECTS)
 
 distclean:	clean
-		rm -f $(SDK_STATIC) $(SDK_DYNAMIC) $(HOSTSDK_STATIC) $(HOSTSDK_DYNAMIC) $(PLUGIN_TARGET) $(HOST_TARGET) $(RDFGEN_TARGET) *~ */*~
+		rm -f $(SDK_STATIC) $(SDK_DYNAMIC) $(HOSTSDK_STATIC) $(HOSTSDK_DYNAMIC) $(PLUGIN_TARGET) $(HOST_TARGET) $(RDFGEN_TARGET) *~ */*~ config.log config.status Makefile
 
 install:	$(SDK_STATIC) $(SDK_DYNAMIC) $(HOSTSDK_STATIC) $(HOSTSDK_DYNAMIC) $(PLUGIN_TARGET) $(HOST_TARGET) $(RDFGEN_TARGET)
 		mkdir -p $(DESTDIR)$(INSTALL_API_HEADERS)
diff -Nur -x '*.orig' -x '*~' debian.git/lib-src/lib-widget-extra/Makefile.in debian.git.new/lib-src/lib-widget-extra/Makefile.in
--- debian.git/lib-src/lib-widget-extra/Makefile.in	2009-06-14 15:40:18.000000000 +0200
+++ debian.git.new/lib-src/lib-widget-extra/Makefile.in	2009-06-14 15:41:54.803169276 +0200
@@ -82,6 +82,9 @@
 	rm -rf $(OBJS)
 	rm -f $(LIBFILE) $(EXTRAS)
 
+distclean: clean
+	rm -rf config.log config.status Makefile
+
 #
 # Rule for compiling C++ files
 #
diff -Nur -x '*.orig' -x '*~' debian.git/lib-src/Makefile.in debian.git.new/lib-src/Makefile.in
--- debian.git/lib-src/Makefile.in	2009-02-03 13:03:56.000000000 +0100
+++ debian.git.new/lib-src/Makefile.in	2009-06-14 15:43:09.287169573 +0200
@@ -222,6 +222,7 @@
 	-$(MAKE) -C soundtouch distclean
 	-$(MAKE) -C twolame distclean
 	-rm -f @LOCAL_LIBS@
+	-rm -f libportmixer.a
 
 # list here all the targets that aren't actually files to be made
 .PHONY: portSMF eexpat-recursive FileDialog-recursive \
