Description: Fix segfault with multiple ALSA devices
Bug-Debian: http://bugs.debian.org/584605
Bug: http://bugzilla.audacityteam.org/show_bug.cgi?id=174
Author: Dave Witbrodt <dawitbro@sbcglobal.net>
Applied-Upstream: 1.3.13

--- a/lib-src/portmixer/src/px_linux_alsa.c
+++ b/lib-src/portmixer/src/px_linux_alsa.c
@@ -136,7 +136,10 @@
                dev->numselems++;
             }
             else if (snd_mixer_selem_is_enum_capture(elem)) {
-               dev->numselems += snd_mixer_selem_get_enum_items(elem);
+               int retval = snd_mixer_selem_get_enum_items(elem);
+
+               if (retval > 0)
+                  dev->numselems += retval;
             }
          }
       }
@@ -196,12 +199,15 @@
             i++;
          }
          else if (snd_mixer_selem_is_enum_capture(elem)) {
-            unsigned int cnt = snd_mixer_selem_get_enum_items(elem);
-            unsigned int j;
+            int j;
+            int cnt = snd_mixer_selem_get_enum_items(elem);
+
+            if (cnt < 0)
+               continue;
 
             for (j = 0; j < cnt; j++) {
                char iname[256];
-               snd_mixer_selem_get_enum_item_name(elem, j, sizeof(iname), iname);
+               snd_mixer_selem_get_enum_item_name(elem, (unsigned int) j, sizeof(iname), iname);
                snprintf(name,
                         sizeof(name),
                         "%s:%d",
