--- configure.in.orig	2008-05-14 14:16:04.000000000 +0100
+++ configure.in	2008-05-14 14:15:27.000000000 +0100
@@ -480,7 +480,20 @@
 
 dnl PortAudio configuration
 
-if [[ "$use_portaudio" = "v19" ]] ; then
+if [[ "$use_portaudio" = "system" ]] ; then
+       AC_CHECK_HEADER([portaudio.h],, AC_MSG_ERROR([Could not find portaudio.h header file.]))
+       AC_CHECK_LIB(portaudio, Pa_GetDeviceCount, [have_pa=v19], [have_pa=no])
+       if [[ "$have_pa19" = "no" ]] ; then
+               AC_CHECK_LIB(portaudio, Pa_CountDevices, [have_pa=v18], [AC_MSG_ERROR([Could not find portaudio library.])])
+       else
+               AC_DEFINE(USE_PORTAUDIO_V19, 1,
+                       [Define if PortAudio v19 is being used (instead of v18)])
+       fi
+
+       LIBS="$LIBS -lportaudio"
+       use_portaudio=$have_pa
+
+elif [[ "$use_portaudio" = "v19" ]] ; then
    dnl PortAudio v19
    AC_DEFINE(USE_PORTAUDIO_V19, 1,
       [Define if PortAudio v19 is being used (instead of v18)])
